File: app/src/main/java/com/example/focusstartsecondpart/features/guests/data/GuestsDataSource.java
package com.example.focusstartsecondpart.features.guests.data;

import com.example.focusstartsecondpart.features.guests.domain.model.Guest;

import java.util.List;

import io.reactivex.Single;
import io.reactivex.SingleObserver;

public interface GuestsDataSource {
    Single<List<Guest>> loadGuests(int id);

    SingleObserver<Guest> updateGuest();

    void updateGuestToNet(int eventId, Guest guest);
}

File: app/src/main/java/com/example/focusstartsecondpart/features/guests/data/GuestsDataSource.kt
package com.example.focusstartsecondpart.features.guests.data

import com.example.focusstartsecondpart.features.guests.domain.model.Guest
import com.example.focusstartsecondpart.features.guests.domain.model.Result
import com.example.focusstartsecondpart.features.guests.domain.model.VerifiedMember

import io.reactivex.Single

interface GuestsDataSource {
    fun loadGuests(id: Int): Single<List<Guest>>
    fun updateGuest(eventId: Int, guest: Guest, verifiedMemberList: List<VerifiedMember>): Single<Result>
}
